ethcore/src/builtin.rs

```diff
@@ -101,18 +101,17 @@ impl Pricer for ModexpPricer {
 		let exp_len = read_len();
 		let mod_len = read_len();
 
+		if mod_len.is_zero() && base_len.is_zero() {
+			return U256::zero()
+		}
+
 		let max_len = U256::from(u32::max_value() / 2);
-		if base_len > max_len || mod_len > max_len {
+		if base_len > max_len || mod_len > max_len || exp_len > max_len {
```

Comment:
does this work properly with base = 1, mod = 1, exp = 2^256-1?

Reply:
@arkpar ^^^

Reply:
Yes, according to the spec it should fail because gas calculation formula would return a huge value.